### PR TITLE
Removed unused WMBS DAOs

### LINKED ISSUE #8301: Unused DAOs to be removed
Related to changes in #8280, these DAOs seem to be used nowhere:
src/python/WMCore/WMBS/MySQL/Jobs/LoadFileLocations.py
src/python/WMCore/WMBS/Oracle/Jobs/LoadFileLocations.py

src/python/WMCore/WMBS/MySQL/Files/GetBulkLocation.py
src/python/WMCore/WMBS/Oracle/Files/GetBulkLocation.py

src/python/WMCore/WMBS/MySQL/Subscriptions/GetAcquiredFilesByLimit

src/python/WMCore/WMBS/MySQL/Subscriptions/GetAvailableFilesByRun
src/python/WMCore/WMBS/MySQL/Subscriptions/GetAvailableFilesByLimit

src/python/WMCore/WMBS/MySQL/Subscriptions/GetCompletedFilesByLimit.py
src/python/WMCore/WMBS/MySQL/Subscriptions/GetCompletedFilesByRun.py

src/python/WMCore/WMBS/MySQL/Subscriptions/GetFailedFilesByLimit.py
src/python/WMCore/WMBS/MySQL/Subscriptions/GetFailedFilesByRun.py

### PR DESCRIPTION
**Pull request by amaltaro**
Fix #8301

unless I messed my greps up, all this code is not used neither in WMCore nor in T0.

### CODE REVIEW

**Comment by amaltaro** [comment]
Haaaa, I knew some of those were being used...
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMBS/Subscription.py#L174

**Comment by ticoann** [comment]
Alan some are used for unittests - getFileLocations and might be used some script in Tier0.
Need to check with Juan, Dirk.

File: src/python/WMCore/WMBS/Job.py [hunk 1]
```diff
@@ -222,20 +222,6 @@ def getFiles(self, type="list"):
         self.commitTransaction(existingTransaction)
         return WMJob.getFiles(self, type)
 
-    def getFileLocations(self):
```
Comment (amaltaro) [inline src/python/WMCore/WMBS/Job.py]: @hufnagel @vytjan folks, is this method used by T0 or by any of your scripts?

Comment (vytjan) [reply src/python/WMCore/WMBS/Job.py]: @amaltaro I don't think it's used anywhere by T0, but I'd rather wait for Dirk to confirm this.

**Comment by amaltaro** [comment]
However, if a DAO is only used for unit tests, I'd rather not have this DAO in the src code and instead write the query itself in the unittest where it's needed. As I did for a couple of cases here.

**Comment by amaltaro** [comment]
test this please

File: src/python/WMCore/WMBS/Job.py (hunk 1 above)
Comment (hufnagel) [reply src/python/WMCore/WMBS/Job.py]: I am on vacation. If you want to know, "grep -r ", on the Tier code :-).

Comment (ticoann) [reply src/python/WMCore/WMBS/Job.py]: I wasn't talking about just this method but I grepped for T0 code. I was just wondering whether there is the ops specific code (not in git repository) using one of the DAOs.
If no one knows, I guess Not. 😄

Comment (hufnagel) [reply src/python/WMCore/WMBS/Job.py]: All code I know about is in the github repo. Tier0Ops would have to comment on scripts they use.

File: src/python/WMCore/WMBS/Subscription.py [hunk 2]
```diff
@@ -182,14 +182,9 @@ def filesOfStatus(self, status, limit=0, loadChecksums=True, doingJobSplitting=F
 
         status = status.title()
         files = set()
-        if limit > 0:
-            action = self.daofactory(classname="Subscriptions.Get%sFilesByLimit" % status)
```
Comment (amaltaro) [inline src/python/WMCore/WMBS/Subscription.py]: @ticoann so, these DAOs could be under use, but we had not a single call to this method with the limit argument, that's why I just got rid of them :)

File: src/python/WMCore/WMBS/Subscription.py [hunk 3]
```diff
@@ -322,26 +317,6 @@ def isCompleteOnRun(self, runID):
...
-        """
-        _filesOfStatusByRun_
-
-        Return all the files in the given subscription and the given run which
-        have the given status.
-        """
-        existingTransaction = self.beginTransaction()
-
-        files = []
-        action = self.daofactory(classname="Subscriptions.Get%sFilesByRun" % status)
```
Comment (amaltaro) [inline src/python/WMCore/WMBS/Subscription.py]: Seangchan, same here, nowhere in WMCore or T0 code we call this method. Meaning, delete it!!!

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
@ticoann now it's failing on unstable tests only. Can you please review and merge it? I'll need this one merged first, then the failure on 8280 should be gone

**Review by ticoann — APPROVED** [review]

**Comment by ticoann** [comment]
test this please

**Pull request merged**